src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -377,25 +376,25 @@ type AccessStrategy<'event, 'state> =
 type private Category<'event, 'state, 'context>(context: MessageDbContext, codec: IEventCodec<_, _, 'context>, fold, initial, access) =
     let loadAlgorithm log category streamId streamName requireLeader ct =
         match access with
-        | None -> context.LoadBatched(log, streamName, requireLeader, codec.TryDecode, ct)
-        | Some AccessStrategy.LatestKnownEvent -> context.LoadLast(log, streamName, requireLeader, codec.TryDecode, ct)
+        | None -> context.LoadBatched(log, streamName, requireLeader, codec.TryDecode, fold, initial, ct)
+        | Some AccessStrategy.LatestKnownEvent -> context.LoadLast(log, streamName, requireLeader, codec.TryDecode, fold, initial, ct)
         | Some (AccessStrategy.AdjacentSnapshots (snapshotType, _)) -> task {
             match! context.LoadSnapshot(log, category, streamId, requireLeader, codec.TryDecode, snapshotType, ct) with
             | ValueSome (pos, snapshotEvent) ->
-                let! token, rest = context.Reload(log, streamName, requireLeader, pos, codec.TryDecode, ct)
-                return token, Array.insertAt 0 snapshotEvent rest
-            | ValueNone -> return! context.LoadBatched(log, streamName, requireLeader, codec.TryDecode, ct) }
-    let fetch state f = task { let! token', events = f in return struct (token', fold state (Seq.ofArray events)) }
-    let reload (log, sn, leader, token, state) ct = fetch state (context.Reload(log, sn, leader, token, codec.TryDecode, ct))
+                let state = fold initial [| snapshotEvent |]
+                let! token, state = context.Reload(log, streamName, requireLeader, pos, codec.TryDecode, fold, state, ct)
+                return struct(token, state)
+            | ValueNone -> return! context.LoadBatched(log, streamName, requireLeader, codec.TryDecode, fold, initial, ct) }
+    let reload (log, sn, leader, token, state) ct = context.Reload(log, sn, leader, token, codec.TryDecode, fold, state, ct)
     interface ICategory<'event, 'state, 'context> with
         member _.Load(log, categoryName, streamId, streamName, _maxAge, requireLeader, ct) =
-            fetch initial (loadAlgorithm log categoryName streamId streamName requireLeader ct)
+            loadAlgorithm log categoryName streamId streamName requireLeader ct
         member x.TrySync(log, categoryName, streamId, streamName, ctx, _maybeInit, token, state, events, ct) = task {
             let encode e = codec.Encode(ctx, e)
             let encodedEvents: IEventData<EventBody>[] = events |> Array.map encode
             match! context.TrySync(log, categoryName, streamId, streamName, token, encodedEvents, ct) with
             | GatewaySyncResult.Written token' ->
-                let state' = fold state (Seq.ofArray events)
```

Comment:
this changes the contact, better to leave for the PR that does that

Reply:
fixed